internal/dag/httpproxy_processor.go

```diff
@@ -1343,6 +1334,53 @@ func (p *HTTPProxyProcessor) computeSecureVirtualHostAuthorization(validCond *co
 	return true
 }
 
+func computeVirtualHostRateLimitPolicy(proxy *contour_api_v1.HTTPProxy, rateLimitService *contour_api_v1alpha1.RateLimitServiceConfig, validCond *contour_api_v1.DetailedCondition) *RateLimitPolicy {
+	rlp := &RateLimitPolicy{}
+	var err error
+
+	// Set the local rateLimit policy
+	if proxy.Spec.VirtualHost.RateLimitPolicy != nil && proxy.Spec.VirtualHost.RateLimitPolicy.Local != nil {
+		rlp.Local, err = localRateLimitPolicy(proxy.Spec.VirtualHost.RateLimitPolicy.Local)
+		if err != nil {
+			validCond.AddErrorf(contour_api_v1.ConditionTypeRouteError, "RateLimitPolicyNotValid",
+				"Spec.VirtualHost.RateLimitPolicy.Local is invalid: %s", err)
+			return nil
+		}
+	}
+
+	// Set the global rateLimit policy if HTTPProxy defines one
+	if proxy.Spec.VirtualHost.RateLimitPolicy != nil &&
+		proxy.Spec.VirtualHost.RateLimitPolicy.Global != nil &&
```

Comment:
can `proxy.Spec.VirtualHost.RateLimitPolicy.Global` be != nil and `en(proxy.Spec.VirtualHost.RateLimitPolicy.Global.Descriptors) == 0` does envoy accept that config?

Reply:
no, rate limit service will fail